src/ConfigurationHelper.php

```diff
@@ -0,0 +1,539 @@
+   *
+   * @param \Drupal\search_api\IndexInterface $index
+   *   The index to the channel field to.
+   */
+  protected function indexAddChannelsField(IndexInterface $index): void {
+    if ($index->getField(Constants::CHANNEL_SELECTION_FIELD)) {
+      return;
+    }
+
+    $field = new SearchIndexField($index, Constants::CHANNEL_SELECTION_FIELD);
+    $field->setLabel('Directory channels');
+    $field->setDataSourceId('entity:node');
+    $field->setPropertyPath(Constants::CHANNEL_SELECTION_FIELD);
+    $field->setType('string');
+    $field->setDependencies([
+      'config' => [
+        'field.storage.node.' . Constants::CHANNEL_SELECTION_FIELD,
+      ],
+    ]);
+    $index->addField($field);
+  }
+
+  /**
+   * Import config entity for the directory Facet.
+   */
+  public function createFacet(): void {
+    if ($this->entityTypeManager->getStorage('facets_facet')->load(Constants::FACET_CONFIG_ENTITY_ID)) {
+      return;
+    }
+
+    $conditional_config_path = $this->moduleExtensionList->getPath('localgov_directories') . '/config/conditional';
+    if ($this->importConfigEntity('facets_facet', $conditional_config_path, Constants::FACET_CONFIG_FILE)) {
+      $this->configInstaller->installOptionalConfig();
+    }
+  }
+
+  /**
+   * Update a block's visibility to add to content type.
+   *
+   * The given block should appear sidebar pages for the given content type.
+   *
+   * @param string $block_id
+   *   The block to update visibility for.
+   * @param string $content_type
+   *   The content type on which the block should be visible.
+   *
+   * @return bool
+   *   True on success.
+   */
+  public function blockAddContentType(string $block_id, string $content_type): bool {
+    $block_config = $this->entityTypeManager->getStorage('block')->load($block_id);
+    if (!$block_config instanceof BlockInterface) {
+      $this->logger->error('Block %block-id is missing.  Cannot update its visibility settings.', [
+        '%block-id' => $block_id,
+      ]);
+
+      return FALSE;
+    }
+
+    try {
+      $visibility = $block_config->getVisibility();
+      $visibility['node_type']['bundles'][$content_type] = $content_type;
+      $block_config->setVisibilityConfig('node_type', $visibility['node_type']);
+      $block_config->save();
+    }
+    catch (\Exception $e) {
+      $this->logger->error('Failed to add %content-type content type to %block-id block: %error-msg', [
+        '%content-type' => $content_type,
+        '%block-id' => $block_id,
+        '%error-msg' => $e->getMessage(),
+      ]);
+
+      return FALSE;
+    }
+
+    $this->logger->notice('Added %content-type content type to %block-id block.', [
+      '%content-type' => $content_type,
+      '%block-id' => $block_id,
+    ]);
+
+    return TRUE;
+  }
+
+  /**
+   * Update a block's visibility to remove from content type.
+   *
+   * The given block should no longer appear sidebar pages for the given
```

Comment:
s/sidebar pages/in the sidebar of pages/

Reply:
Corrected in https://github.com/localgovdrupal/localgov_directories/pull/233/commits/b187781ad0067d98be3ee637f3ef155c6c967ba2

Reply:
Thanks.